apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/ClaimVerification/ClaimVerificationDialog.tsx

```diff
@@ -0,0 +1,269 @@
+import { useIntl } from "react-intl";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { useMutation } from "urql";
+import { useState } from "react";
+
+import {
+  ClaimVerificationResult,
+  Maybe,
+  Scalars,
+  UpdatePoolCandidateClaimVerificationInput,
+  graphql,
+} from "@gc-digital-talent/graphql";
+import { toast } from "@gc-digital-talent/toast";
+import { Button, Dialog } from "@gc-digital-talent/ui";
+import { DateInput, RadioGroup } from "@gc-digital-talent/forms";
+import { errorMessages, formMessages } from "@gc-digital-talent/i18n";
+import { strToFormDate } from "@gc-digital-talent/date-helpers";
+
+import { priorityTitle, veteranTitle } from "./messages";
+
+type FormValues = {
+  result: ClaimVerificationResult;
+  expiry?: Scalars["DateTime"]["input"];
+};
+
+const UpdateClaimVerification_Mutation = graphql(/* GraphQL */ `
+  mutation UpdatePriorityVerification(
+    $id: UUID!
+    $values: UpdatePoolCandidateClaimVerificationInput!
+  ) {
+    updatePoolCandidateClaimVerification(id: $id, poolCandidate: $values) {
+      id
+    }
+  }
+`);
+
+const todayDate = new Date();
+
+interface ClaimVerificationDialogProps {
+  context: "veteran" | "priority";
+  id: Scalars["UUID"]["output"];
+  result?: Maybe<ClaimVerificationResult>;
+  expiry?: Maybe<Scalars["DateTime"]["output"]>;
+  priorityNumber?: Maybe<string>;
+}
+
+const ClaimVerificationDialog = ({
+  id,
+  context,
+  result,
+  expiry,
+  priorityNumber,
+}: ClaimVerificationDialogProps) => {
+  const intl = useIntl();
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [{ fetching }, executeMutation] = useMutation(
+    UpdateClaimVerification_Mutation,
+  );
+  const isPriority = context === "priority";
+
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      expiry: expiry ?? undefined,
+      result: result ?? undefined,
+    },
+  });
+
+  const verificationResult = methods.watch("result");
+
+  const handleError = () => {
+    toast.error(
+      intl.formatMessage({
+        defaultMessage: "Error: could not update claim verification",
+        id: "8HMXsd",
+        description:
+          "Error message for updating a candidates claim verification",
+      }),
+    );
+  };
+
+  const handleSubmit: SubmitHandler<FormValues> = (data) => {
+    const values: UpdatePoolCandidateClaimVerificationInput = isPriority
+      ? {
+          priorityVerification: data.result,
+          priorityVerificationExpiry:
+            data.result === ClaimVerificationResult.Accepted
+              ? data?.expiry
+              : null,
+        }
+      : {
+          veteranVerification: data.result,
+          veteranVerificationExpiry: data?.expiry ?? null,
+        };
```

Comment:
Maybe ought to null out dates when you submit a non-ACCEPTED input. Nothing breaking, but probably good to avoid odd data states.

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/d73d9dcf-2f34-4e6f-beb0-4b33c8261d0d)

From setting as ACCEPTED with an expiration date, then updating to UNVERIFIED for whatever reason. (Accidental or say you got new information)

Reply:
Good catch, fixed in [only pass expiry when veteran claim accepted](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10502/commits/bfb3d42b46b9f4e630e694dec2f2157df49feb77)